src/matrix/kaldi-vector.h

```diff
@@ -310,6 +310,10 @@ class VectorBase {
   /// Returns sum of the elements
   Real Sum() const;
 
+  /// Returns sum of the elements raised to power p. For p>0, this is the same
+  /// as Norm(p) raised to the p-th power.
+  Real SumPower(Real p) const;
```

Comment:
The documentation says that it's the same as Norm(p)^p (except for p == 0, where it seems to be the same).
That being said, I don't see why it's needed.  Can't you just replace its invocation wth Norm(p)^p?